src/HTTPResponse.cpp

```diff
@@ -56,16 +61,91 @@ void HTTPResponse::handleRequestGET(const HTTPRequest& request, const ServerConf
 	}
 }
 
+std::string HTTPResponse::extractFolderName(const std::string& uri) {
+	if (uri.empty())
+		return ("");
+	size_t lastSlashPos;
+	if (uri[uri.size() - 1] == '/')
+		lastSlashPos = uri.find_last_of('/', uri.length() - 2);
+	else
+		lastSlashPos = uri.find_last_of('/');
+	if (lastSlashPos == std::string::npos || lastSlashPos == uri.length() - 1)
+		return ("");
+	size_t start = lastSlashPos + 1;
+	size_t end = uri.find('/', start);
+	if (end == std::string::npos)
+		end = uri.length();
+	return (uri.substr(start, end - start));
+}
+
 void HTTPResponse::serveFile(const ServerConfig& serverConfig, const std::string& uri) {
-	std::string filePath = serverConfig.rootDirectory + (uri == "/" ? "/index.html" : uri);
-	std::ifstream file(filePath.c_str());
-	INFO("Serving file: " << filePath);
-	if (file) {
-		std::string content((std::istreambuf_iterator<char>(file)), std::istreambuf_iterator<char>());
-		assignResponse(200, content, determineContentType(uri));
-		file.close();
+	std::string fullPath = serverConfig.rootDirectory + uri;
```

Comment:
Can you break up this function into helper functions? The function is around 70 lines of code and is hard to read